src/cljam/io/bcf/reader.clj

```diff
@@ -252,19 +282,43 @@
    (read-variants rdr {}))
   ([^BCFReader rdr {:keys [depth] :or {depth :deep}}]
    (.seek ^BGZFInputStream (.reader rdr) ^long (.start-pos rdr))
-   (let [contigs (meta->map (:contig (.meta-info rdr)))
-         filters (assoc (meta->map (:filter (.meta-info rdr))) 0 {:id "PASS" :kw :PASS})
-         formats (meta->map (:format (.meta-info rdr)))
-         info (meta->map (:info (.meta-info rdr)))
-         kws (mapv keyword (drop 8 (.header rdr)))
-         parse-fn (case depth
-                    :deep (comp (partial bcf-map->parsed-variant contigs filters formats info kws)
-                                parse-data-line-deep)
-                    :vcf (comp (vcf-util/variant-vals-stringifier (.meta-info rdr) (.header rdr))
-                               (partial bcf-map->parsed-variant contigs filters formats info kws)
-                               parse-data-line-deep)
-                    :bcf parse-data-line-deep
-                    :shallow (partial parse-data-line-shallow contigs)
-                    :raw identity)]
+   (let [info (meta->map (:info (.meta-info rdr)))
+         parse-fn  (make-parse-fn rdr info depth)]
      (read-data-lines (.reader rdr)
                       (fn [rdr] (parse-fn (read-data-line-buffer rdr)))))))
+
+(defn- make-lazy-variants [f s]
+  (when-first [fs s]
+    (lazy-cat
+     (f fs)
+     (make-lazy-variants f (rest s)))))
+
+(defn read-variants-randomly
+  "Reads variants of the bgzip compressed BCF file randomly using csi file.
+   Returning them as a lazy sequence."
```

Comment:
Just a matter of English grammar, but it sounds more natural to me if the second sentence starts with `Returns them as ...` instead of `Returning them as ...`, unless you join these two sentences into one:

```suggestion
  Returns them as a lazy sequence."
```